modules/mcuboot/CONFIG_BUILD_WITH_TFM.conf

```diff
@@ -0,0 +1,7 @@
+# This Kconfig fragment is applied to the MCUBoot configuration when
+# we are building for TF-M. This is necessary so MCUBoot can configure
+# itself to boot TF-M correctly.
+
+CONFIG_FPROTECT=n
```

Comment:
This is used to ensure that the memory area used by mcuboot is not overwritten. Its not clear to me why this should conflict with TF-M. Do you have details on why this must be disabled? If so we should probably comment that here.

Reply:
@sigvartmh  ?

Reply:
Pretty sure this was added because it didn't work without I don't think I did enough investigation into why. I'll double-check if it is needed. Could be some misconfiguration somewhere that causes this to fail. Seems like it has worked at some point from my chat log with Bernt.

Reply:
My suspicion is that TF-M assumes that the access control HW it uses is at reset value, in this case the SPU.

That was the case with the MPU at least. It assumed that the MPU was configured such at all memory access was unpriviliged, but due to configuration from MCUBoot, it was configured such that all memory access was privileged.

IMHO it is reasonable for the SPE to do all access control.

But I am not certain.

Reply:
Would it make sense for TF-M to verify that a region is not already protected before it tries to apply protection to it? I guess that would resolve this issue? Adding this would make it more robust IMO. We could make a ticket for that and then leave this here for now with a comment.

Reply:
@SebastianBoe Do you know where in MCUBoot it sets this access? Is it the MPU_ALLOW_FLASH_WRITE? It tries to clean up the MPU but maybe something is missing https://github.com/nrfconnect/sdk-mcuboot/blob/b514a3190821b7721dd6ffcd9940474ebe7a8832/boot/zephyr/arm_cleanup.c#L30

Aligning MCUBoot on 32KB or 16KB aka SPU region fixes this issue and you should be able to boot it. At least I was able to successfully boot two applications with FPROTECT enabled. This should probably be resolved in partition manager so that it enforces 32/16 K alignment of Primary slot. Also the check mentioned above would be nice.

Reply:
I don't know where MCUBoot configures the MPU.

As I see it, it is more sensible to avoid configuring something, than to configure it and then unconfigure it.

I don't want to fix the z_arm_clear_arm_mpu_config function because according to it's documentation it intends only to clear the region configuration, which it is doing successfully.

I don't know if it should do something else, and therefore don't want to change what it's doing.

Reply:
> Would it make sense for TF-M to verify that a region is not already protected before it tries to apply protection to it? I guess that would resolve this issue? Adding this would make it more robust IMO. We could make a ticket for that and then leave this here for now with a comment.

This issue proved to be about global access control wrt. the MPU and alignment wrt. SPU. So I don't think this applies.

We could perhaps create an issue looking like

"Investigate what assumptions and requirements TF-M has about how the MPU and SPU is configured prior to SPE being booted. And verify that these requirements are enforced."

If that fits what you had in mind?